Introduce global, per-tenant flags and interval to evaluation threshold to control rule evaluation concurrency (#8146)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

# CLI flag: -ruler.protected-namespaces
[ruler_protected_namespaces: <string> | default = ""]

# (experimental) Maximum number of independent rules that can run concurrently
# for each tenant. Depends on `-ruler.max-global-rule-evaluation-concurrency`

> # being more than 0 and ideally this flag should be a lower value. 0 to disable.

Reviewer: ```suggestion
# being greater than 0. Ideally, this flag should be a lower value. Set this value to `0` to disable.
```

Author: In https://github.com/grafana/mimir/pull/8146/commits/de2e543694f5c8b66da5465e01bf58b46b161424 I have taken the first two suggestions.

I have not incorporated the change of "0 to disable" because this is consistent with the rest of the flags we have. If this is something you think we should do you can change it on the `limits.go` file and I'll be happy to help you get it in.
